### DISCUSSION TITLE
Contribution guide

### DISCUSSION THREAD

**Discussion by siennathesane** [nxrighthere/UnrealCLR#90] [General]
This is an awesome toolkit, I'm really excited to dive into it and start using it. Looking at your roadmap and what's available now, there are some APIs I'm interested in using sooner rather than later (such as networking and async). I was looking around, but I didn't see a contribution guide or any hints on how or where to contribute APIs to increase the surface area. Is there a contribution guide on how to contribute to this project?

Also, congrats on the Epic Mega Award. :)

**Comment by nxrighthere** [comment]
Hello and thank you for your interest in contributing to the project!

Unfortunately, the process of functionality implementation and exposure is way too complicated for contribution from 3rd-parties. I made a specific toolset that handles code generation automatically, so I only have to do some cleanup and verification of code on my side.

On top of this, I also have a specific vision of how things should be done here for general purpose usage, so I don't expect contributions of code from 3rd-parties (see https://github.com/nxrighthere/UnrealCLR/discussions/11).

The best contribution for me is feedback from users.

> Also, congrats on the Epic Mega Award. :)

Thank you!

**Comment by siennathesane** [comment]
Ah that's a bummer. I've done a lot of work for Go to improve the codegen tooling for Windows, so I have lots of experience with cross-language codegens so I was interested in helping. If you can find a way to implement tooling and documentation for what you have, I'm sure there's a lot of folks in the community with skills and interest to help.

**Comment by nxrighthere** [comment]
The problem is that the tooling itself is under a different license and it's based on some of my components that I can't disclose.
